cmd/thanos/rule.go

```diff
@@ -377,7 +390,8 @@ func runRule(
 		}
 		fanoutStore := storage.NewFanout(logger, agentDB, remoteStore)
 		appendable = fanoutStore
-		queryable = fanoutStore
+		// Use a separate queryable to restore the ALERTS firing states.
+		queryable = thanosrules.NewPromClientsQueryable(logger, queryClients, promClients, conf.query.httpMethod, conf.query.step, conf.restoreIgnoreLabels)
```

Comment:
I'm wondering here why do we need a prom clients queryable? Shouldn't using just `remoteStore` work fine?

Reply:
Not sure if I understand this correctly...
But remote store doesn't ensure that we can query it and remote read must be implemented.
But in our case, thanos receiver doesn't implement remote read.
We have two choices: 1. Use prom clients to query Thanos querier.
2. Implement remote read at Thanos Receiver.

The first choice makes more sense to me as Ruler needs to use only one source for alerting and state restore.

Reply:
Thanks for the explanation, I was looking at the code incorrectly and I thought we can also read from remote store, but that's not true. Then it makes sense to proceed with querier :+1:.